docs/source/developers/release.rst

```diff
@@ -216,123 +216,365 @@ Be sure to go through on the following checklist:
 #. Bump versions
 #. Update tags for Go modules
 #. Update docs
+#. Announce the new release
 #. Remove old artifacts
 
-.. dropdown:: Marking the released version as "RELEASED" on JIRA
+.. dropdown:: Mark the released version as "RELEASED" on JIRA
    :animate: fade-in-slide-down
    :class-title: sd-fs-5
    :class-container: sd-shadow-md
 
-    Open https://issues.apache.org/jira/plugins/servlet/project-config/ARROW/administer-versions
+   - Open https://issues.apache.org/jira/plugins/servlet/project-config/ARROW/administer-versions
+   - Click "..." for the release version in "Actions" column
+   - Select "Release"
+   - Set "Release date"
+   - Click "Release" button
 
-    Click "..." for the release version in "Actions" column
+.. dropdown:: Start the new version on JIRA
+   :animate: fade-in-slide-down
+   :class-title: sd-fs-5
+   :class-container: sd-shadow-md
+
+   - Open https://issues.apache.org/jira/plugins/servlet/project-config/ARROW/administer-versions
+   - Click "..." for the next version in "Actions" column
+   - Select "Edit"
+   - Set "Start date"
+   - Click "Save" button
+
+.. dropdown:: Merge changes on release branch to maintenance branch for patch releases
+   :animate: fade-in-slide-down
+   :class-title: sd-fs-5
+   :class-container: sd-shadow-md
 
-    Select "Release"
+   Merge ``release-X.Y.Z-rcN`` to ``maint-X.Y.Z``:
 
-    Set "Release date"
+   .. code-block:: Bash
 
-    Click "Release" button
+      # git checkout maint-10.0.0
+      git checkout maint-X.Y.Z
+      # git merge release-10.0.0-rc0
+      git merge release-X.Y.Z-rcN
+      # git push -u apache maint-10.0.0
+      git push -u apache maint-X.Y.Z
 
-.. dropdown:: Starting the new version on JIRA
+.. dropdown:: Add the new release to the Apache Reporter System
    :animate: fade-in-slide-down
    :class-title: sd-fs-5
    :class-container: sd-shadow-md
 
-    Open https://issues.apache.org/jira/plugins/servlet/project-config/ARROW/administer-versions
+   Add relevant release data for Arrow to `Apache reporter <https://reporter.apache.org/addrelease.html?arrow>`_.
 
-    Click "..." for the next version in "Actions" column
+.. dropdown:: Upload source release artifacts to Subversion
+   :animate: fade-in-slide-down
+   :class-title: sd-fs-5
+   :class-container: sd-shadow-md
 
-    Select "Edit"
+   A PMC member must commit the source release artifacts to Subversion:
 
-    Set "Start date"
+   .. code-block:: Bash
 
-    Click "Save" button
+      # dev/release/post-01-upload.sh 0.1.0 0
+      dev/release/post-01-upload.sh <version> <rc>
 
-.. dropdown:: Updating the Arrow website
+.. dropdown:: Upload binary release artifacts to Artifactory
    :animate: fade-in-slide-down
    :class-title: sd-fs-5
    :class-container: sd-shadow-md
 
-    Fork the `arrow-site repository <https://github.com/apache/arrow-site>`_ and clone it next to the arrow repository.
+   A PMC member must upload the binary release artifacts to Artifactory:
```

Comment:
I thought committers also have access to the artifactory, not entirely sure though, might need to request the access, couldn't find anything in the infra wiki. :shrug:

Reply:
Right. I've fixed.